### PR TITLE
Added sendgrid integration

### PR DESCRIPTION
**Pull request by ruipoliveira**

### CODE REVIEW

File: backend/fief/services/email/sendgrid.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        text: Optional[str] = None,
+    ):
+        from_email, from_name = sender
+        to_email, to_name = recipient
+        try:
+            message = Mail(
+                from_email=format_address(from_email, from_name),
+                to_emails=format_address(to_email, to_name),
+                subject=subject,
+                html_content=text)
```
Comment (frankie567) [inline backend/fief/services/email/sendgrid.py:29]: `html_content` argument should be assigned to the `html` variable. I [see there is a `plain_text_content`](https://github.com/sendgrid/sendgrid-python/blob/653cc47c54d53d401d65d1debb3cdc1d7905b4a1/sendgrid/helpers/mail/mail.py#L23-L32) that should be good for `text`.

So we should have:

```py
message = Mail(
    from_email=format_address(from_email, from_name),
    to_emails=format_address(to_email, to_name),
    subject=subject,
    html_content=html,
    plain_text_content=text,
)
```

File: backend/fief/services/email/sendgrid.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+            self._client.send(message)
+
+        except Exception as e:
```
Comment (frankie567) [inline backend/fief/services/email/sendgrid.py:33]: Could we tighten up the exception catch here? Maybe `SendGridException`?

**Review by frankie567 — CHANGES_REQUESTED** [review]
Good work! Added a small comments in the implementation. One other thing: you need to run `hatch run lint` to format the code and pass the linters.

File: backend/fief/services/email/sendgrid.py (hunk 1 above)
Comment (ruipoliveira) [reply backend/fief/services/email/sendgrid.py]: Ah, yes make sense.

File: backend/fief/services/email/sendgrid.py (hunk 2 above)
Comment (ruipoliveira) [reply backend/fief/services/email/sendgrid.py]: Yes, is correct. Thank you

**Comment by ruipoliveira** [comment]
> Good work! Added a small comments in the implementation. One other thing: you need to run `hatch run lint` to format the code and pass the linters.

It's done! I wait for your feedback
Thanks @frankie567

**Comment by frankie567** [comment]
LGTM 👍 

@all-contributors add @ruipoliveira for code

**Pull request merged**